src/VisualStudio/Core/Def/Implementation/LanguageClient/LiveShareLanguageServerClient.cs

```diff
@@ -0,0 +1,103 @@
+        /// </summary>
+        public string Name => ServicesVSResources.LiveShare_CSharp_Visual_Basic_Language_Server_Client;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public IEnumerable<string>? ConfigurationSections { get; } = null;
+
+        /// <summary>
+        /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
+        /// This may be null if the client does not need custom initialization options.
+        /// </summary>
+        public object? InitializationOptions { get; } = null;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public IEnumerable<string>? FilesToWatch { get; } = null;
+
+        public event AsyncEventHandler<EventArgs>? StartAsync;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public event AsyncEventHandler<EventArgs>? StopAsync { add { } remove { } }
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public LiveShareLanguageServerClient(LanguageServerProtocol languageServerProtocol, VisualStudioWorkspace workspace, IDiagnosticService diagnosticService, IThreadingContext threadingContext)
+        {
+            _languageServerProtocol = languageServerProtocol;
+            _workspace = workspace;
+            _diagnosticService = diagnosticService;
+            _threadingContext = threadingContext;
+        }
+
+        public Task<Connection> ActivateAsync(CancellationToken token)
+        {
+            var (clientStream, serverStream) = FullDuplexStream.CreatePair();
+            _ = new InProcLanguageServer(serverStream, serverStream, _languageServerProtocol, _workspace, _diagnosticService, _threadingContext);
+            return Task.FromResult(new Connection(clientStream, clientStream));
+        }
+
+        /// <summary>
+        /// Signals that the extension has been loaded.  The server can be started immediately, or wait for user action to start.  
+        /// To start the server, invoke the <see cref="StartAsync"/> event;
+        /// </summary>
+        public async Task OnLoadedAsync()
+        {
+            await StartAsync.InvokeAsync(this, EventArgs.Empty).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Signals the extension that the language server has been successfully initialized.
+        /// </summary>
+        public Task OnServerInitializedAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Signals the extension that the language server failed to initialize.
+        /// </summary>
+        public Task OnServerInitializeFailedAsync(Exception e)
+        {
+            return Task.CompletedTask;
```

Comment:
this seems like something we need to know about.

Reply:
Updated comment here to explain why it just returns completed task.